src/commands/appSettings/connectionSettings/durableTaskScheduler/DTSConnectionSetSettingStep.ts

```diff
@@ -6,17 +6,21 @@
 import { nonNullProp } from '@microsoft/vscode-azext-utils';
 import { ConnectionKey } from '../../../../constants';
 import { SetConnectionSettingStepBase } from '../SetConnectionSettingStepBase';
-import { type IDTSConnectionWizardContext } from './IDTSConnectionWizardContext';
+import { type IDTSAzureConnectionWizardContext, type IDTSConnectionWizardContext } from './IDTSConnectionWizardContext';
 
 export class DTSConnectionSetSettingStep<T extends IDTSConnectionWizardContext> extends SetConnectionSettingStepBase<T> {
     public priority: number = 240;
     public debugDeploySetting: ConnectionKey = ConnectionKey.DTS;
 
     public async execute(context: T): Promise<void> {
-        await this.setConnectionSetting(context, nonNullProp(context, 'newDTSConnection'));
+        let newDTSConnectionSetting = nonNullProp(context, 'newDTSConnectionSetting');
+        if ((context as unknown as IDTSAzureConnectionWizardContext).managedIdentity) {
+            newDTSConnectionSetting = newDTSConnectionSetting.replace('<ClientID>', (context as unknown as IDTSAzureConnectionWizardContext).managedIdentity?.clientId ?? '');
```

Comment:
I understand sometimes we just gotta make the typings happy, but this is pretty uggo. Why not just have the context be 
IDTSConnectionWizardContext | IDTSAzureConnectionWizardContext or Partial<IDTSAzureConnectionWizardContext>?

Reply:
Or since `IDTSAzureConnectionWizardContext` extends `IDTSConnectionWizardContext` anyway, just replace it?

Reply:
Although type-wise using `IDTSAzureConnectionWizardContext` works fine, I think it implies that all connections that use that step for Azure related, which isn't the case.  I prefer your original one for the accuracy, although it's a little more verbose: `IDTSConnectionWizardContext | IDTSAzureConnectionWizardContext`

Unfortunately I still have to do the typecasting this way, but at least I can omit the `as unknown` which I agree looks pretty bad!